pkg/client/connector/intercept.go

```diff
@@ -127,6 +129,41 @@ func (tm *trafficManager) workerPortForwardIntercepts(ctx context.Context) error
 	return nil
 }
 
+// startForwards starts port forwards and mounts for the given forwardKey. It returns true if either was started (port-forward or a mount), false otherwise
+func (tm *trafficManager) startForwards(ctx context.Context, wg *sync.WaitGroup, fk forwardKey, sftpPort int32, extraPorts []int32) bool {
+	started := false
+	if sftpPort > 0 {
+		// There's nothing to mount if the SftpPort is zero
+		started = true
+		mntCtx := dgroup.WithGoroutineName(ctx, fmt.Sprintf("/%s:%d", fk.PodIP, sftpPort))
+		wg.Add(1)
+		go tm.workerMountForwardIntercept(mntCtx, mountForward{fk, sftpPort}, wg)
+	}
+	for _, port := range extraPorts {
+		started = true
+		pfCtx := dgroup.WithGoroutineName(ctx, fmt.Sprintf("/%s:%d", fk.PodIP, port))
+		wg.Add(1)
+		go tm.portForward(pfCtx, wg, fk, port)
+	}
+	return started
+}
+
+func (tm *trafficManager) portForward(ctx context.Context, wg *sync.WaitGroup, fk forwardKey, port int32) {
+	defer wg.Done()
+	// Using kubectl port-forward here would require the pod name to be either fetched from the API server, or threaded
+	// all the way through from the intercept request to the agent and into the WatchIntercepts. Instead, we re-use
+	// the forwarder from the agent, and dial the pod's IP directly.
+	addr := net.TCPAddr{
+		IP:   net.IPv4(127, 0, 0, 1),
+		Port: int(port),
+	}
+	f := agent.NewForwarder(&addr, fk.PodIP, port)
```

Comment:
This feels like a violation of implicit package structure rules. A `pkg` should not access things under `cmd`. If the `Forwarder` no longer is agent specific, then please move it to a package under `pkg`.